aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -164,7 +166,10 @@ module aptos_framework::code {
         };
 
         // Request publish
-        request_publish(addr, module_names, code, policy.policy)
+        if (version::current() < version::next())
```

Comment:
I'm not quite following what this is for even after reading the comments in version.move. Also there's a TODO left there to update the value in next(). Can you explain (and add some comments here) what this is for and how this works?

Reply:
There are multiple discussions on slack about this design pattern, plus a notion doc, plus documentation with example in `version`, and finally some Diem history. Please let me concretely know what you are missing. The discussion about this pattern is already in a late state, I'll be happy to bring you up to speed.

Reply:
Yeah. It might be good to add either more comments to remind readers why this is happening. It might also be good to have a sync so we can really ingrain into our minds what the problem and risks are. We're all going 100 mph in different directions, so it is easy to forget.

Reply:
> not clear why allowed deps is toggled based upon version changing and not sure what the tests are testing. what would a failure look like?

Because those allowed deps are breaking change which require a new native function they need to be version protected. 

The tests check that the code behaves correctly regards downwards compatibility. The new Move and the native code must work both for the previous and the next version.